create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/MainHandlerTest.java

```diff
@@ -107,17 +108,17 @@ public void testOkResponse()
 
         mainHandler.handleRequest(mainHandlerInputStream(), output, context);
 
-        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        GatewayResponse<?> gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
```

Comment:
You can parse the GatewayResponse without the use of wildcards here. See suggestion. Also, we should try to keep the same level of abstraction in one method. Json parsing is an implementation detail is a level lower of what we are testing here and should be abstracted by usage of private methods.

```suggestion
 @Test
    public void testOkResponse()
        throws Exception {

        MainHandler mainHandler = createMainHandler(environment);
        ByteArrayOutputStream output = new ByteArrayOutputStream();
        mainHandler.handleRequest(mainHandlerInputStream(), output, getMockContext());

        GatewayResponse<Summary> gatewayResponse = parseSuccessResponse(output.toString());
        assertEquals(SC_OK, gatewayResponse.getStatusCode());
        assertTrue(gatewayResponse.getHeaders().containsKey(CONTENT_TYPE));
        assertTrue(gatewayResponse.getHeaders().containsKey(MainHandler.ACCESS_CONTROL_ALLOW_ORIGIN));

        Summary summary = gatewayResponse.getBodyObject(Summary.class);
        assertNotNull(summary.getIdentifier());
    }
    
      private GatewayResponse<Summary> parseSuccessResponse(String output) throws JsonProcessingException {
        return parseGatewayResponse(output, Summary.class);
    }

    private GatewayResponse<Problem> parseFailureResponse(String output) throws JsonProcessingException {
        return parseGatewayResponse(output, Problem.class);
    }

    private <T> GatewayResponse<T> parseGatewayResponse(String output, Class<T> responseObjectClass)
        throws JsonProcessingException {
        JavaType typeRef = objectMapper.getTypeFactory()
            .constructParametricType(GatewayResponse.class, responseObjectClass);
        return objectMapper.readValue(output, typeRef);
    }


```